libsnark/gadgetlib1/gadgets/curves/weierstrass_g2_gadget.tcc

```diff
@@ -46,6 +46,19 @@ G2_variable<ppT>::G2_variable(protoboard<FieldT> &pb,
     all_vars.insert(all_vars.end(), Y->all_vars.begin(), Y->all_vars.end());
 }
 
+template<typename ppT>
+G2_variable<ppT>::G2_variable(protoboard<FieldT> &pb,
+                              const Fqe_variable<ppT> &X_coord,
+                              const Fqe_variable<ppT> &Y_coord,
+                              const std::string &annotation_prefix) :
+    gadget<FieldT>(pb, annotation_prefix),
+    X(new Fqe_variable<ppT>(pb, X_coord.c0, X_coord.c1, FMT(annotation_prefix, " X"))),
+    Y(new Fqe_variable<ppT>(pb, Y_coord.c0, Y_coord.c1, FMT(annotation_prefix, " Y")))
+{
```

Comment:
This constructor does seem to break the genericity of this gadget as it will only make sense in the context where twist elements are built over a field extensions of degree 2 (i.e. `Fqe_variable` needs to have 2 coordinates for this function to be well defined).
As of now, the constructor from a `const libff::G2<other_curve<ppT> >` was keeping the gadget fully generic (e.g. works with https://github.com/clearmatics/libff/blob/develop/libff/algebra/curves/bls12_377/bls12_377_g2.hpp#L43 and https://github.com/clearmatics/libff/blob/develop/libff/algebra/curves/bw6_761/bw6_761_g2.hpp#L37 which does not seem to be the case for this extra G2 gadget constructor)

Reply:
Thanks.  Implemented in terms of the Fqe_variable copoy cnostructor, so it should be generic now.

Reply:
Perfect, thanks